java-generator/core/src/test/java/io/fabric8/java/generator/RoundTripTest.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.fabric8.kubernetes.client.Version;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import org.apache.maven.shared.invoker.*;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class RoundTripTest {
+    private static TemporaryFolder tmpFolder = TemporaryFolder.builder().build();
```

Comment:
btw https://junit.org/junit5/docs/current/user-guide/#writing-tests-built-in-extensions-TempDirectory

Reply:
My main reason for keeping this "custom" implementation is that, during development, I often replace the `tmpFolder.delete()` with a `System.out.println(tmpFolder.getAbsolutePath())` to check and debug the generated code. Do we have the opportunity to do the same with the built-in functionality?

Reply:
It's possible to get the absolute path of both java.nio.file.Path or java.io.File, yes

Reply:
... and avoid to remove the content?

Reply:
Yup

`FileChannel.open(path, StandardOpenOption.WRITE).truncate(0).close();`

Reply:
I struggle to see how it works, my bad, do you wanna open a PR so that we can talk on code?

Reply:
I want the `TempDir` to NOT be deleted(sometimes in dev mode) so that I can inspect the generated files.